### PR TITLE
chore: Run tests as make commands in workflows

### PR DESCRIPTION
**Pull request by tokoko**
**What this PR does / why we need it**:
As some workflows are marked as `pull_request_target`, it's hard to change the pytest command to run tests in a single PR. This PR changes workflows to run tests as make commands.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -64,24 +64,20 @@ benchmark-python-local:
...
-	python -m pytest -n 8 sdk/python/tests \
+	python -m pytest -n 8 --color=yes sdk/python/tests
 
 test-python-integration:
-	FEAST_USAGE=False IS_TEST=True python -m pytest -n 8 --integration sdk/python/tests
+	python -m pytest -n 8 --integration --color=yes --durations=5 --timeout=1200 --timeout_method=thread sdk/python/tests
 
 test-python-integration-local:
 	@(docker info > /dev/null 2>&1 && \
-		FEAST_USAGE=False \
```
Comment (sudohainguyen) [inline Makefile:76]: I believe we should keep `FEAST_USAGE` and `IS_TEST`

File: Makefile [hunk 2]
```diff
@@ -64,24 +64,20 @@ benchmark-python-local:
 	FEAST_USAGE=False IS_TEST=True FEAST_IS_LOCAL_TEST=True python -m pytest --integration --benchmark  --benchmark-autosave --benchmark-save-data sdk/python/tests
 
 test-python:
```
Comment (sudohainguyen) [inline Makefile:66]: consider renaming it to `test-python-unit` ?

**Review by sudohainguyen — CHANGES_REQUESTED** [review]

File: Makefile (hunk 1 above)
Comment (tokoko) [reply Makefile:76]: I found that there's being set in [conftest](https://github.com/feast-dev/feast/blob/master/sdk/python/tests/conftest.py#L27) regardless

File: Makefile (hunk 2 above)
Comment (tokoko) [reply Makefile]: good idea. done

File: Makefile (hunk 1 above)
Comment (tokoko) [reply Makefile:76]: fixed, thanks

**Review by sudohainguyen — APPROVED** [review]
lgtm

**Comment by tokoko** [comment]
@franciscojavierarceo fyi, had to make some changes to `stream_feature_view`, typecheck was complaining about `body_text` of type string being passed to `feature_transformation` parameter. Not sure how this was missed in ci previously though, something about bumping library versions, i suppose.

**Comment by jeremyary** [comment]
This one's been soaking for a few days for feedback now. LGTM, I support pulling commands up and out of the action via leveraging make. Thanks!

**Pull request merged**